### ISSUE TITLE
[Bug]: 16157-1 is too lax about simple types

### ISSUE THREAD

**Issue by JosefKaltwasser** [DATEX-II-EU/DatexII#517]
### What happened?

16157-1 gives only very short definitions about D2Datatypes. These do no clearly specify the lexical and/or value space. The schemaType properties used for the one PSM already covered in 16157-1 (XML) maps those constructs to something appearing suitable in XSD. As an example for what can happen look at the schemaType of D2Datatype URL in 16157-7, which is mapped to xs:anyURI. This type covers all URLs, but also many things which are not a valid URL.
It would be better if PIM definitions refer to standards - or explicitly specified lexical and value space itself - without reference to potential PSMs. E.g. for the D2Datatype "URL", the definition should refer to the BNF specification of a URL in RFC 3986. The PSMs should describe how they implement the corresponding concepts, e.g. the mapping to xs:anyURI, which makes the 'error' described above.
16157-1 would also have to specify how manipulation of the lexical/value space via facets is affecting datatype inheritance. It is assumed - but not explicitly stated - that a subtype inherits the schema type including all manipulations.
This should be addressed in v4, since there the whole problem now assembles in 16157-1.

### Version

4

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by VictoriaSteinerAC** [comment]
In order to rephrase the definitions we had a look at the current definitions and the two PSMs for XSD and JSON schema, and created a table with three sections (see attachment at the bottom of the comment):
1.	DATEX II generic data types (current definition, proposed new definition, comment)
2.	XSD mapping (current mapping to XSD (and definition of the data type from XSD), comment, valid and invalid examples)
3.	JSON schema mapping (current mapping to JSON schema (and definition of the data type from JSON schema), options for update of current mapping, comment, valid and invalid examples)

In general, we tried to analyse three things:
1.	Are the definitions of the generic datatypes in Part 1 sufficient (i.e. clearly specify the lexical and/or value space or give a reference to respective specifications where the data type is specified)
2.	Is the mapping to XSD exactly fitting the DATEX II definition of the respective data type. If no, the differences should be defined in the mapping.
3.	Is the mapping to JSON schema exactly fitting the DATEX II definition of the respective data type. If no, the differences should be defined in the mapping. Also, are there any other possibilities to enhance the mapping, e.g. by using patterns or formats in JSON schema?

[DATEX II Generic Datatypes.xlsx](https://github.com/user-attachments/files/16437877/DATEX.II.Generic.Datatypes.xlsx)

**Comment by iancornwellmottmac** [comment]
Comments on the updated proposals for PIM definitions:
- Boolean - the added sentence is redundant and I don't think it should be added. I think it is always true that a PSM mapping can map things including lexical values. By saying this here we may imply you couldn't have done it if we hadn't said it.
- Date - OK, that is probably quite a practical change, choosing a standard platform-independent lexical encoding rather than stopping at the semantics like the old definition. It does force binary PSMs to adopt that lexical approach, rather than some built-in binary date that they might support, but I suppose the benefit is greater than any theoretical drawback. (there is a stray copy paste error in the comment field though - from Base64Binary)
- Double - I don't like the proposed change. I can see the old definition supports both lexical and binary representations. I haven't read the IEEE 754-2019 standard but I'm assuming that its binary64 format specifies a 64-bit binary encoding, which I think is the wrong focus for a standard typically implemented in lexical encodings. It seems dubious to me say that a lexically expressed double complies with binary64. I assume that's why XSD itself says "patterned after".  However, rather than totally rejecting the proposed change, I can see that we may want to give some standardised meaning to values like "NaN", which the current definition does not, so some compromise may be possible with a different wording to show that it's the semantics of IEEE 754 and not the encoding that is important.
- Float - similar comment to Double.

**Comment by VictoriaSteinerAC** [comment]
- Boolean: Josef and I agree, we go back to the original definition.
- Date: Regarding the second sentence, this was not really our intention. Thus we propose to add the sentence "Binary PSMs may opt to use a binary encoding for the lexical space, but the value space must adhere to the definitions provided by ISO 8601.". The same would be added for DateTime and Time.
- Double: Yes we agree. Character base representations should not be excluded. The intention was to have a reference to a standardised definition of the value space. Thus we could change the definition to "A double-precision 64-bit floating point value in the binary64 format as specified in IEEE 754-2019. Since IEEE 754-2019 only defines a binary encoding, the PSM for character based plattforms may specify an appropriate encoding of the lexical space. However, the value space must adhere to the definitions provided by IEEE 754-2019."
- Float: same for double.

Feel free to propose improved definitions!

**Comment by JosefKaltwasser** [comment]
Linked to #501

**Comment by iancornwellmottmac** [comment]
If further inspiration is sought for definitions, consider consulting ISO/IEC 11404 "Information Technology - General-Purpose Datatypes", a revision of which is currently out for ballot as DIS 11404:2024 through ISO/IEC JTC1/SC32. I am not suggesting we rewrite any of our datatype definitions to refer to that, but if we are considering change to definitions anyway, we might align with definitions there, if we are satisfied that they also align well to the relevant XML and JSON schema types.

**Comment by JosefKaltwasser** [comment]
Agreed version has been incorporated into German response to Enquiry.
